desktop/src/com/frostwire/gui/bittorrent/TorrentFetcherDownload.java

```diff
@@ -242,7 +245,11 @@ private void downloadTorrent(final byte[] data, final List<TcpEndpoint> peers) {
             try {
                 TorrentInfo ti = TorrentInfo.bdecode(data);
                 boolean[] selection = calculateSelection(ti, relativePath);
-                BTEngine.getInstance().download(ti, null, selection, peers);
+                if (ConnectionSettings.MANDATORY_VPN_FOR_BITTORRENT.getValue() && !VPNs.isVPNActive()) {
+                    GUIMediator.showWarning(I18n.tr("VPN is inactive. Current settings require active VPN connection before starting BitTorrent engine. Setup a VPN connection or check advanced settings."));
+                } else {
+                    BTEngine.getInstance().download(ti, null, selection, peers);
+                }
```

Comment:
so I see a pattern here with this `if/else`, let's refactor this into a method that takes all the parameters as the `BTEngine.getInstance().download()` method takes on the else. #DRY